avm/res/recovery-services/vault/backup-config/main.bicep

```diff
@@ -0,0 +1,96 @@
+metadata name = 'Recovery Services Vault Backup Config'
+metadata description = 'This module deploys a Recovery Services Vault Backup Config.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Conditional. The name of the parent Azure Recovery Service Vault. Required if the template is used in a standalone deployment.')
+param recoveryVaultName string
+
+@description('Optional. Name of the Azure Recovery Service Vault Backup Policy.')
+param name string = 'vaultconfig'
+
+@description('Optional. Enable this setting to protect hybrid backups against accidental deletes and add additional layer of authentication for critical operations.')
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+param enhancedSecurityState string = 'Enabled'
+
+@description('Optional. ResourceGuard Operation Requests.')
+param resourceGuardOperationRequests array = []
+
+@description('Optional. Enable this setting to protect backup data for Azure VM, SQL Server in Azure VM and SAP HANA in Azure VM from accidental deletes.')
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+param softDeleteFeatureState string = 'Enabled'
+
+@description('Optional. Storage type.')
+@allowed([
+  'GeoRedundant'
+  'LocallyRedundant'
+  'ReadAccessGeoZoneRedundant'
+  'ZoneRedundant'
+])
+param storageModelType string = 'GeoRedundant'
+
+@description('Optional. Storage type.')
+@allowed([
+  'GeoRedundant'
+  'LocallyRedundant'
+  'ReadAccessGeoZoneRedundant'
+  'ZoneRedundant'
+])
+param storageType string = 'GeoRedundant'
+
+@description('Optional. Once a machine is registered against a resource, the storageTypeState is always Locked.')
+@allowed([
+  'Locked'
+  'Unlocked'
+])
+param storageTypeState string = 'Locked'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+@description('Optional. Is soft delete feature state editable.')
+param isSoftDeleteFeatureStateEditable bool = true
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
```

Comment:
In AVM, the telemetry is defined in the parent module only. Please remove the defaultTelemetry references from all child modules. Also the related stuff, like parameter enableDefaultTelemetry etc.